src/dune_engine/build_system.ml

```diff
@@ -434,7 +434,13 @@ end = struct
            without sandboxing. We just need to make sure we clean up all stale
            directory targets before running the rule and then we can discover
            all created files right in the build directory. *)
-        if not (Path.Build.Set.is_empty targets.dirs) then
+        if
```

Comment:
As the above comment says, we might need to do some preparation before running a non-sandboxed action that produces directory targets. What if those directories already contain some files, for example?